tools/src/wyvern/target/corewyvernIL/decl/Declaration.java

```diff
@@ -10,6 +10,16 @@
 import wyvern.tools.errors.FileLocation;
 
 public abstract class Declaration extends ASTNode implements EmitOIR {
+	private boolean hasResource = false;
+
+	public boolean containsResource() {
+		return this.hasResource;
+	}
+
+	public void setHasResource(boolean hasResource) {
```

Comment:
The AST is nearly all immutable (the major exception is ObjectValue).  Let's not create a public setter for hasResource (here or elsewhere).